tcl/org.testeditor.tcl.dsl/src/org/testeditor/tcl/dsl/Tcl.xtext

```diff
@@ -12,81 +12,111 @@
  *******************************************************************************/
 grammar org.testeditor.tcl.dsl.Tcl with org.eclipse.xtext.xbase.Xbase
 
-
-
 import "http://www.testeditor.org/tcl"
 import "http://www.testeditor.org/tsl" as tsl
 import "http://www.testeditor.org/aml" as aml
 import "http://www.eclipse.org/emf/2002/Ecore" as ecore
 
-
 TclModel:
 	'package' package=QualifiedName
 	(importSection=XImportSection)?
-	test=TestCase
-;
+	test=TestCase;
 
-TestCase: 
+TestCase:
 	'#' name=ID
 	('implements' specification=[tsl::TestSpecification|QualifiedName])?
-	steps+=SpecificationStepImplementation*
-;
-
+	steps+=SpecificationStepImplementation*;
 
-SpecificationStepImplementation: 
+SpecificationStepImplementation:
 	{SpecificationStepImplementation}
 	'*' contents+=StepContent* '.'?
-	contexts+=TestStepContext*
-;
+	contexts+=TestStepContext*;
 
 TestStepContext:
 	('Mask' | 'Component') ':' component=[aml::Component]
-	steps+=(TestStep | TestStepWithAssignment | TestStepAssertion )*
-;
+	steps+=(TestStep | TestStepWithAssignment | AssertionTestStep)*;
 
 TestStep:
 	{TestStep}
-	'-' contents+=StepContent* '.'?
-;
-
-TestStepAssertion:
-	'-' 'assert' (negated?='!')? variableName=ID (comparator=Comparator contents+=StepContent)? '.'?
-;
+	'-' contents+=StepContent* '.'?;
+
+AssertionTestStep:
+	'-' 'assert' expression=AssertionExpression '.'?;
+
+/** Assertion Exression is either a null check, 
+ *  or a full comparison that requires a comparator and a right expression (in order to have no ambiguity with the null check), 
+ *  or an inner expression in parenthesis (no more null checks allowed in here!) 
+ */
+AssertionExpression:
+	AENullCheck | AEFullComparison | '(' AssertionInnerExpression ')';
```

Comment:
We should refactor the class names of the subclasses of `AssertionExpression`. But not necessarily now.